proto/sequencerblockapis/astria/sequencerblock/v1/block.proto

```diff
@@ -151,6 +151,24 @@ message RollupData {
   }
 }
 
+message UpgradeChangeHashesWithProof {
+  // The SHA256 digests of all upgrade changes applied during this block, if an upgrade was
+  // activated at this height.
+  repeated bytes upgrade_change_hashes = 1;
+  // The proof that the change hashes are included in the cometbft block data (if it exists),
+  // specifically the third item in the data field.
+  astria.primitive.v1.Proof proof = 2;
```

Comment:
why do we need an upgrade merkle proof? the reason for including a proof for the other fields is that it's needed by conductor (using only DA) to verify the data as it's relevant to rollups, but I don't think upgrades on the sequencer application are necessarily relevant to rollups/conductor?

Reply:
>but I don't think upgrades on the sequencer application are necessarily relevant to rollups/conductor

I agree for this first upgrade, but I'm not sure that that will always be the case?  I don't think we have a detailed plan yet for how to handle a breaking change which _would_ affect the conductor (e.g. we change how txs are encoded).

Also, I think it's worthwhile since any external consumer could validate the upgrade via the DA, not necessarily just the conductor.

Saying that, this could be added later relatively easily I guess (not for historical upgrades of course), and including it does cost, in terms of computation and storage fees.  So maybe it's worth discussing further offline?

Reply:
I think this falls into YAGNI, as long as the general upgrade mechanism exists we could add this functionality if we have a need in future.

Reply:
Removed in [d21fef9](https://github.com/astriaorg/astria/pull/1917/commits/d21fef98a61ef646becae6b8fd547d78ea836f75).